### PR TITLE
feat: on_validation_error callback

### PR DESCRIPTION
**Pull request by falconandy**

### CODE REVIEW

File: sdk/sig/itly.rbs [hunk 1]
```diff
@@ -1,6 +1,4 @@
 class Itly
-  include Itly::Plugins
```
Comment (justin-fiedler) [inline sdk/sig/itly.rbs:2]: @falconandy why remove this?

File: sdk/spec/ilty/itly_spec.rb [hunk 2]
```diff
@@ -216,6 +216,48 @@
...
+          method: :identify, pass_data_as: :properties,
+          method_params: { user_id: '123',
+                           properties: { data: '1', info: 'yes' },
+                           options: { 'fake_plugin0' => FakeCallOptions.new(data: 'for plugin 0') } },
+          validation_value: Itly::Options::Validation::DISABLED,
+          set_on_validation_error: true,
+          expected_event_properties: { data: '1', info: 'yes' },
+          expected_log_info: 'identify(user_id: 123, properties: {:data=>"1", :info=>"yes"}, '\
+            'options: {"fake_plugin0"=>#<FakeCallOptions: for plugin 0>})',
+          generate_validation_error: true, expect_validation: false
```
Comment (justin-fiedler) [inline sdk/spec/ilty/itly_spec.rb:231]: Does this mean `on_validation_error` will be called in the `Validation::DISABLED` case?

File: sdk/sig/itly.rbs (hunk 1 above)
Comment (falconandy) [reply sdk/sig/itly.rbs:2]: With the line I have a error:
```
$ bin/rspec --format documentation --order defined

An error occurred while loading spec_helper.
Failure/Error: end

RBS::MixinClassError:
  ./sig/itly.rbs:2:2...2:23: Cannot include a class `::Itly::Plugins` in the definition of `::Itly`
```

The `Itly::Plugins` specification is empty:
```
class Itly
  class Plugins
  end
end
```

File: sdk/spec/ilty/itly_spec.rb (hunk 2 above)
Comment (falconandy) [reply sdk/spec/ilty/itly_spec.rb:231]: It means `on_validation_call` is not expected in the Validation::DISABLED case.
 
https://github.com/amplitude/itly-sdk-ruby/blob/DXOC-591-on-validation-error/sdk/spec/support/shared_examples/validate_and_run_on_plugins.rb#L97-L104

**Review by justin-fiedler — APPROVED** [review]

**Pull request merged**